CMakeLists.txt

```diff
@@ -0,0 +1,12 @@
+cmake_minimum_required(VERSION 3.4)
+
+set(zenfs_SOURCES "fs/fs_zenfs.cc" "fs/zbd_zenfs.cc" "fs/io_zenfs.cc" PARENT_SCOPE)
+set(zenfs_LIBS "zbd" PARENT_SCOPE)
+set(zenfs_CMAKE_EXE_LINKER_FLAGS "-u zenfs_filesystems_reg" PARENT_SCOPE)
+
+execute_process(WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+                RESULT_VARIABLE GENVER_RESULT
+                COMMAND ./generate-version.sh)
+if(GENVER_RESULT AND NOT GENVER_RESULT EQUAL 0)
+  message(FATAL_ERROR "Generating ZenFS version failed")
+endif()
```

Comment:
Is there also a way to add a check for the correct libzbd version like here? :

https://github.com/westerndigitalcorporation/zenfs/blob/master/zenfs.mk#L12

Reply:
I did not find a way to do that. I'll investigate it a bit more.

Reply:
It is possible to add the version check when running cmake, so I've added that. Updating the rocksdb.pc file won't be part of this change.